doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1183,357 @@ processed in this order:
 - If the value is a sequence, its elements are recursively imported in the
   order they appear.
 
+.. _west-manifest-groups:
+
+Project Groups and Active Projects
+**********************************
+
+You can use the ``groups`` keys briefly described :ref:`above
+<west-manifest-files>` to place projects into groups. The ``groups`` key can be
+a project or top level manifest attribute, like this:
```

Comment:
When reading this, a user might think that the `groups` key is is used for same purpose, placing a project in a group, regardless of whether the `groups` key is placed at project or manifest level.

But the fact is, that the `groups` key at project level describes to which groups a project belongs, and at manifest level it specifies what groups to en-/disable.

Could you re-phrase this ?
Sorry for not having a proposal here.

Reply:
> But the fact is, that the groups key at project level describes to which groups a project belongs, and at manifest level it specifies what groups to en-/disable.

I find it unfortunate that these two different keys have the same name.

Reply:
> I find it unfortunate that these two different keys have the same name.

Would you like to propose an alternative? We haven't released this yet. I'm open to suggestions.

Reply:
Ok, let me give this a try, feel free to improve further:

>  At project level the ``groups`` defines the groups to which the project belong, at the top level manifest attribute the ``groups`` enable or disable groups, like this:

Reply:
> Ok, let me give this a try, feel free to improve further:

@tejlmand I think @marc-hb is saying he would like to rename "manifest: groups:" to "manifest: something-different-than-groups:".

Reply:
ok.
but I still think it would be valuable for the reader if it was highlighted that the two `groups` key are used for different purposes.
But this is just minor, as it will be clear in later sections.

Reply:
How about `manifest.projects.project.in_groups`. Besides avoiding the collision with the other key, `in_groups` also reduces IMHO the potential confusion between `- groupA` in this YAML list and `-groupA` elsewhere, I mean imagine someone completely new to west, then the `in_` prefix makes it IMHO harder to read too fast and mistake for a `-` exclusion:
```
  in_groups:
   - groupA
```


For the other key, how about: `manifest.groups_filter` as in Python list comprehension etc.

Reply:
> How about `manifest.projects.project.in_groups`

I'm not in favor of this change. At the API level, a `west.manifest.Project` has a `groups` attribute which corresponds to this data, so making this key `groups` is concise and consistent.

> For the other key, how about: manifest.groups_filter as in Python list comprehension etc.

I'm OK with something like this.

I guess you meant this to apply to both the configuration file and the manifest. This also impacts the `west update --groups` option.

I'd like to use `groups-filter` instead of `groups_filter` for consistency with other things (like `west-commands`, `repo-path`).

------------------------------------

**Summary of proposed change**

1. In YAML:

```yaml
manifest:
  projects:
     - name: myproject
       groups: # stays the same
         - foo
         - bar
  groups-filter: [-foo,-bar] # changes from 'groups' to 'groups-filter'
```

2. In config: the `manifest.groups` configuration option becomes `manifest.groups-filter`.
3. In command line options: the new `--groups` option in `west update` becomes `--groups-filter`.

------------------------------------

Thoughts from @marc-hb  @carlescufi @tejlmand and others?

Reply:
> I'm not in favor of this change.

OK, as long as one of them is renamed then they will differ from each other which I think is the main goal. That's why I tried to give options for either.

> I guess you meant this to apply to both the configuration file and the manifest. 

I didn't look that far but sure.

> I'd like to use groups-filter instead of groups_filter for consistency with other things (like west-commands, repo-path).

+1

Reply:
I think `groups` as now is fine, but I also think `groups-filter` is just as good.

No preference for one or the other.

Reply:
As long as we rename this to `group-filter`, I am OK with renaming the manifest-wide option, alongside the config and command-line ones.